### DISCUSSION TITLE
single-command installation

### DISCUSSION THREAD

**Discussion by zjin-lcf** [AdaptiveCpp/AdaptiveCpp#634] [General]
Is there single-command installation for tool developers and users ?  Could you provide a single-command install script ? 

For example, users download Nvidia HPC SDK and then type ./install to install the SDK. 

Thanks

**Comment by illuhad** [comment]
hipSYCL is a flexible platform that can tie into multiple other compilers and backends. Depending on your system setup, these dependencies can live in a wide variety of locations, so I don't think we can provide a script that does the right thing *for all use cases*. Some users only want the OpenMP backend. Others want the HIP backend in which case ROCm must be installed etc.

You can use our binary packages from our package repositories if you have a supported distribution, and then install e.g. with `apt install`.
Some install scripts are available in `install/scripts`.
There is a 3rd party spack package for hipSYCL.

A basic installation of hipSYCL from source requires no more than `cmake -DCMAKE_INSTALL_PREFIX=/path/to/install /path/to/source && make install`.
I don't see how an installation script could substantially simplify this? For more complex use cases you would have to pass some arguments to this install script anyway to tell it where to find dependencies.

The NVIDIA HPC SDK is in an easier position, because it only supports one type of platform and is a monolithic closed-source collection of binaries that just need to be deployed.

CC @sbalint98
